src/requests/walletrequests.cpp

```diff
@@ -0,0 +1,172 @@
+
+    WalletCreateAccountResponse out;
+
+    out.Result = 0;
+
+    output.load(out);
+
+    return Status::Ok;
+}
+
+Status walletRestoreAccountRequestHandler
+ (const Router::vars_t& vars, 
+  const graft::Input&   input,
+  graft::Context&       context,
+  WalletManager&        wallet_manager,
+  graft::Output&        output)
+{
+    WalletRestoreAccountRequestJsonRpc request;
+
+    if (!input.get(request)) {
+        return errorInvalidParams(output);
+    }
+
+    wallet_manager.restoreAccount(context, request.params.Password, request.params.Seed, getCallbackString(input));
+
+    WalletRestoreAccountResponse out;
+
+    out.Result = 0;
+
+    output.load(out);
+
+    return Status::Ok;
+}
+
+Status walletBalanceRequestHandler
+ (const Router::vars_t& vars, 
+  const graft::Input&   input,
+  graft::Context&       context,
+  WalletManager&        wallet_manager,
+  graft::Output&        output)
+{
+    WalletBalanceRequestJsonRpc request;
+
+    if (!input.get(request)) {
+        return errorInvalidParams(output);
+    }
+
+    wallet_manager.requestBalance(context, request.params.WalletId, request.params.Account, request.params.Password, getCallbackString(input));
+
+    WalletBalanceResponse out;
+
+    out.Result = 0;
+
+    output.load(out);
+
+    return Status::Ok;
+}
+
+Status walletPrepareTransferRequestHandler
+ (const Router::vars_t& vars, 
+  const graft::Input&   input,
+  graft::Context&       context,
+  WalletManager&        wallet_manager,
+  graft::Output&        output)
+{
+    WalletPrepareTransferRequestJsonRpc request;
+
+    if (!input.get(request)) {
+        return errorInvalidParams(output);
+    }
+
+    WalletManager::TransferDestinationArray destinations;
+
+    destinations.reserve(request.params.Destinations.size());
+
+    for (const auto& dest : request.params.Destinations)
+        destinations.push_back(WalletManager::TransferDestination(dest.Address, std::stoull(dest.Amount)));
+
+    wallet_manager.prepareTransfer(context, request.params.WalletId, request.params.Account, request.params.Password, destinations, getCallbackString(input));
+
+    WalletPrepareTransferResponse out;
+
+    out.Result = 0;
+
+    output.load(out);
+
+    return Status::Ok;
+}
+
+namespace
+{
+
+typedef std::function<Status (const Router::vars_t&, const Input&, Context&, WalletManager&, Output&)> RequestHandler;
+
+Router::Handler makeRequestHandler(WalletManager& wallet_manager, const RequestHandler& fn)
+{
+    WalletManager* manager = &wallet_manager;
+    return Router::Handler([fn, manager](const Router::vars_t& vars, const Input& in, Context& context, Output& out) { return fn(vars, in, context, *manager, out); });
+}
+
+void registerWalletRequest(Router& router, WalletManager& wallet_manager, const char* path, int method, const RequestHandler& handler)
+{
+    Router::Handler3 h3(nullptr, makeRequestHandler(wallet_manager, handler), nullptr);
+
+    router.addRoute(path, method, h3);
```

Comment:
You can also do: 
router.addRoute(path, method, {nullptr, makeRequestHandler(wallet_manager, handler), nullptr} );
Up to you

Reply:
I prefer to split this code in several lines. IMO, it's better from readability point of view.